src/models/Settings.js

```diff
@@ -223,19 +227,55 @@ function (Okta, Q, Errors, BrowserFeatures, Util, Logger, config) {
         },
         cache: true
       },
-      // checks if there are any valid configured idps.
-      socialAuthConfigured: {
-        deps: ['configuredSocialIdps'],
-        fn: function (idps) {
-          return !_.isEmpty(idps);
+      // social auth buttons order - 'above'/'below' the primary auth form (boolean)
+      socialAuthPositionTop: {
+        deps: ['configuredSocialIdps', 'idpDisplay'],
+        fn: function (configuredSocialIdps, idpDisplay) {
+          return !!(!_.isEmpty(configuredSocialIdps) && idpDisplay.toUpperCase() === 'PRIMARY');
         },
         cache: true
       },
-      // social auth buttons order - 'above'/'below' the primary auth form (boolean)
-      socialAuthPositionTop: {
-        deps: ['socialAuthConfigured', 'idpDisplay'],
-        fn: function (socialAuthConfigured, idpDisplay) {
-          return !!(socialAuthConfigured && idpDisplay.toUpperCase() === 'PRIMARY');
+      //merges social auth and custom buttons into one array
+      configuredButtons: {
+        deps: ['configuredSocialIdps', 'customButtons'],
+        fn: function (configuredSocialIdps, customButtons) {
+          var self = this;
+          var buttonArray = [];
+          _.each(configuredSocialIdps, function (idp) {
+            var type = idp.type.toLowerCase();
+            var dataAttr = 'social-auth-' + type + '-button';
+            var socialAuthButton = {
+              id: idp.id,
+              type: idp.type,
+              dataAttr: dataAttr,
+              className: 'social-auth-button ' + dataAttr,
+              title: idp.title = Okta.loc('socialauth.' + type + '.label'),
+              click: function (e) {
+                e.preventDefault();
+                OAuth2Util.getTokens(self, {idp: idp.id});
+              }
+            };
+            buttonArray.push(socialAuthButton);
+          });
+          if(_.isArray(customButtons)) {
```

Comment:
Do we actually need the check? I.e. I'm guessing our validator should handle it, and the default value is an empty array.